runtime/domain.c

```diff
@@ -398,8 +438,25 @@ int caml_reallocate_minor_heap(asize_t wsize)
       (void*)domain_self->minor_heap_area,
       (char*)domain_state->young_end - (char*)domain_state->young_start);
 
+  domain_state->young_start =
+    domain_state->young_end =
+    domain_state->young_ptr = NULL;
+  atomic_store_rel(&domain_state->young_limit,
+                   (uintnat) domain_state->young_start);
+}
+
+int caml_allocate_minor_heap(asize_t wsize) {
+  caml_domain_state* domain_state = Caml_state;
+
+  caml_check_minor_heap();
+
   wsize = caml_norm_minor_heap_size(wsize);
 
+  if (wsize > caml_minor_heap_max_wsz) wsize = caml_minor_heap_max_wsz;
```

Comment:
I'm wondering when this branch could be taken.

Reply:
Good question, I am not sure. It may be that we now always update the max minor heap size in the case where this branch would have been taken in the past. This would be a nice simplification, thanks!

Reply:
I think so, yes, changing to the assert.